packages/trpc/server/routers/viewer/bookings/reportBooking.handler.ts

```diff
@@ -0,0 +1,155 @@
+import { BookingReportRepository } from "@calcom/features/bookings/lib/booking-report.repository";
+import { extractBookerEmail } from "@calcom/features/bookings/lib/booking-report.utils";
+import handleCancelBooking from "@calcom/features/bookings/lib/handleCancelBooking";
+import logger from "@calcom/lib/logger";
+import { BookingRepository } from "@calcom/lib/server/repository/booking";
+import prisma from "@calcom/prisma";
+import { BookingStatus } from "@calcom/prisma/enums";
+import type { TrpcSessionUser } from "@calcom/trpc/server/types";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TReportBookingInputSchema } from "./reportBooking.schema";
+
+type ReportBookingOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TReportBookingInputSchema;
+};
+
+const log = logger.getSubLogger({ prefix: ["reportBookingHandler"] });
+
+export const reportBookingHandler = async ({ ctx, input }: ReportBookingOptions) => {
+  const { user } = ctx;
+  const { bookingId, reason, description, cancelBooking, allRemainingBookings } = input;
+
+  const bookingRepo = new BookingRepository(prisma);
+  const bookingReportRepo = new BookingReportRepository(prisma);
+
+  const hasAccess = await bookingRepo.doesUserIdHaveAccessToBooking({
+    userId: user.id,
+    bookingId,
+  });
+
+  if (!hasAccess) {
+    throw new TRPCError({ code: "FORBIDDEN", message: "You don't have access to this booking" });
+  }
+
+  const booking = await bookingRepo.getBookingForReporting({ bookingId });
+
+  if (!booking) {
+    throw new TRPCError({ code: "NOT_FOUND", message: "Booking not found" });
+  }
+
+  // Prevent reporting cancelled or rejected bookings
+  if (booking.status === BookingStatus.CANCELLED || booking.status === BookingStatus.REJECTED) {
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: "Cannot report cancelled or rejected bookings",
+    });
+  }
+
+  const userReport = booking.reports.find((r) => r.reportedById === user.id);
+  if (userReport) {
+    throw new TRPCError({ code: "BAD_REQUEST", message: "You have already reported this booking" });
+  }
```

Comment:
I believe if a booking is already reported (by anyone) we should just flag that as reported. It shouldn't have to be reported by the same user for it to show as reported. Any host of the booking should see it as reported once one of the hosts reported it.